api/utils/cfApiClient.js

```diff
@@ -178,10 +178,26 @@ class CloudFoundryAPIClient {
     { attempt = 1, totalAttempts = 3, sleepInterval = 1000 } = {}
   ) {
     // construct the task parameter template by filling in values from the BuildTaskType metadata
-    // TODO: link to template documentation
+    // https://github.com/cloud-gov/pages-cf-build-tasks/?tab=readme-ov-file#docker-command
     const template = parse(task.BuildTaskType.metadata.template);
 
-    const encryptedTask = Encryptor.encryptObjectValues(task, encryption.key);
+    // prior to adding to the template, SiteBuildTask metadata needs to be combined
+    // with the global rules and stringified. we add this to a separate object to
+    // avoid parameter assignment on the task (in case of retries)
+    const taskToEncrypt = JSON.parse(JSON.stringify(task));
+    const rulesForType = DEFAULT_SCAN_RULES[appMatch(taskToEncrypt.BuildTaskType)];
```

Comment:
Do we need to stringify these as well?

Reply:
This first stringify is really just for doing deep clone, the stringify below actually runs over the combined rulesets